Exclude inactive electronic portfolios/ update electronic_portfolio_s logic

christinach opened the pull request:
closes #1576 
closes #1589

christinach commented:
fixes bad inactive portfolio links that appear in the search results.
Fixes empty available online section in orangelight.

![Screen Shot 2021-08-05 at 5 51 47 PM](https://user-images.githubusercontent.com/9905193/128426149-cf996cb9-701c-42a8-921b-7c4f922f290a.png)
![Screen Shot 2021-08-05 at 5 51 34 PM](https://user-images.githubusercontent.com/9905193/128426152-9c6346b6-dfa5-4e57-a27b-3a034db1c2ee.png)

christinach commented:
vs what is in the catalog with the inactive portfolios:

![Screen Shot 2021-08-05 at 5 40 03 PM](https://user-images.githubusercontent.com/9905193/128426333-ef22a9bd-aeef-479a-85cc-3bee7b8dc868.png)
![Screen Shot 2021-08-05 at 5 38 24 PM](https://user-images.githubusercontent.com/9905193/128426338-07fda575-b67b-47ec-a984-fdb7c0aacba7.png)

tpendragon commented on marc_to_solr/lib/traject_config.rb:
```diff
@@ -1352,7 +1352,9 @@
-  # dont check for scsb
+  # Don't index if 951$a is 'Not Available' - inactive portfolio
+  # Don't check for scsb
+  next if inactive_electronic_portfolio(record)
```
Is there ever a time when one of the 951s is inactive, but other ones aren't? If so, I wonder if this should go down in 1359 similar to `next unless alma_951(record)`?

christinach replied:
yes there is this possibility. let me check

christinach replied:
I'll try to find a fixture with active and inactive

christinach reviewed (commented):
@mzelesky reviewed the pr and I need to do some changes to make it also more efficient. I will also add a custom fixture that has active and inactive portfolios.

christinach commented:
@mzelesky if you could review this pr with me tomorrow. I updated and refactored the code. I'm including also a fixture that has active portfolios and a custom inactive one.

christinach commented on marc_to_solr/lib/traject_config.rb:
```diff
@@ -1352,21 +1352,11 @@
-    embargoes << field
-  end
+  # Don't index if 951$a is 'Not Available' - inactive portfolio
+  # Don't check for scsb
```
@mzelesky I also have a question for the dont check for scsb rule. It was not added. Do I want to add it?

christinach replied:
@mzelesky said it's ok not to add it.

christinach commented:
![Screen Shot 2021-08-09 at 4 52 00 PM](https://user-images.githubusercontent.com/9905193/128772890-3237b56f-2171-4e9b-b530-1b7d726e7e94.png)

vs the empty Available online section:

![Screen Shot 2021-08-09 at 4 53 12 PM](https://user-images.githubusercontent.com/9905193/128773002-52c39c4c-b8ff-4e92-92e3-3bd953410a32.png)

christinach commented:
In this example: https://catalog.princeton.edu/catalog/9996338243506421 the available online section appears empty because the electronic_portfolio url of the sibling record https://catalog.princeton.edu/catalog/9995002873506421/raw does not include a title. It is empty. What are we doing when the title is empty? Do we display the url or a part of it?
The 951 $8 indicates that the sibling_record is an alma record. 
I will review with @mzelesky the 856 of the sibling record to make sure it is indexed as expected.

christinach commented:
Following up on this comment https://github.com/pulibrary/bibdata/pull/1579#issuecomment-895655636
We reviewed with @mzelesky various alma records and different electronic portfolios. A new issue is that there are active portfolios but with no title. The reason is that the existing electronic_portfolio_s logic depends only in an existing collection. We should update the logic with a new condition for cases where the portfolio is part of a book and not a collection. I will open a different issue for that.

hackartisan commented on marc_to_solr/lib/traject_config.rb:
```diff
@@ -1352,21 +1352,10 @@
   fields = []
```
Should these be deleted? they're being set again directly below.

christinach commented:
With the change in #1589 we have 'Full Text' as a title for electronic portfolios that are not part of a collection.:

![Screen Shot 2021-08-10 at 4 11 30 PM](https://user-images.githubusercontent.com/9905193/128928437-dc79c2ca-2170-4fca-a534-f5199748d4bf.png)

christinach commented:
Left side is from catalog-qa (the fixed), right is from production catalog.
https://catalog-qa.princeton.edu/catalog/99123430173506421
https://catalog-qa.princeton.edu/catalog?utf8=%E2%9C%93&search_field=all_fields&q=lux+magazine
https://catalog-qa.princeton.edu/catalog/9996338243506421

![lux_magazine](https://user-images.githubusercontent.com/9905193/129018235-a9a98a54-2365-43fe-87da-9ee567c8a8ce.gif)

![full_text](https://user-images.githubusercontent.com/9905193/129018216-f349c651-6e4f-4d25-98c5-1b12b51be0e8.gif)
